src/compose/application-manager.ts

```diff
@@ -1081,16 +1081,15 @@ export async function getState() {
 			.concat(stateFromServices);
 
 	// Get the list of commits for all appIds from the database
-	const commitsForApp = (
-		await Promise.all(
-			// Deduplicate appIds first
-			[...new Set(servicesToReport.map((svc) => svc.appId))].map(
-				async (appId) => ({
-					[appId]: await commitStore.getCommitForApp(appId),
-				}),
-			),
-		)
-	).reduce((commits, c) => ({ ...commits, ...c }), {});
```

Comment:
Just to note that `.reduce((commits, c) => ({ ...commits, ...c }), {})` is `O(N^2)`

Reply:
Yeah, but the size of `getCommitForApp(appId)` and the number of apps is usually 1, so `O(N^2) = O(1)` @thgreasi

Reply:
In this case yes, but as a general rule it's worth knowing and for things like services/networks/volumes it will be >1 even in the supervisor. Although it's definitely true that we come from the server side where these differences can be amplified much more easily due to the number of requests we see